tests/system/python/conftest.py

```diff
@@ -410,6 +410,294 @@ def _read_data_from_pi(host, admin, password, pi_database, asset, sensor):
+    headers = {'Authorization': 'Basic %s' % username_password_b64, 'Cache-Control': 'no-cache'}
+    normal_header = {'Authorization': 'Basic %s' % username_password_b64}
+    try:
+        conn = http.client.HTTPSConnection(host, context=ssl._create_unverified_context())
+        conn.request("GET", '/piwebapi/assetservers', headers=normal_header)
+        res = conn.getresponse()
+        assert res.status == 200, "Could not request asset server of Pi Web API."
+        r = json.loads(res.read().decode())
+        dbs = r["Items"][0]["Links"]["Databases"]
+
+        if dbs is not None:
+            conn.request("GET", dbs, headers=normal_header)
+            res = conn.getresponse()
+            assert res.status == 200, "Databases is not accessible."
+            r = json.loads(res.read().decode())
+            for el in r["Items"]:
+                if el["Name"] == pi_database:
+                    url_elements_list = el["Links"]["Elements"]
+
+        print("Getting old cache")
+        conn = http.client.HTTPSConnection(host, context=ssl._create_unverified_context())
+        conn.request("GET", '/piwebapi/system/cacheinstances', headers=normal_header)
+        res = conn.getresponse()
+        r = json.loads(res.read().decode())
+        try:
+            # assuming we have single user Administrator
+            old_cache_refresh_time = r["Items"][0]['LastRefreshTime']
+        except (IndexError, KeyError):
+            print("The cache does not exist. ")
+            return
+
+        print("Old cache refresh time {} ".format(old_cache_refresh_time))
+        conn.close()
+
+        print("Going to request element list with cache control no cache.")
+        if url_elements_list is not None:
+            conn.request("GET", url_elements_list, headers=headers)
+            res = conn.getresponse()
+            r = json.loads(res.read().decode())
+
+        conn.close()
+
+        # for verification whether we are able to clear cache.
+        print("Now verifying whether cache cleared.")
+        conn = http.client.HTTPSConnection(host, context=ssl._create_unverified_context())
+        conn.request("GET", '/piwebapi/system/cacheinstances', headers=normal_header)
+        res = conn.getresponse()
+        r = json.loads(res.read().decode())
+        try:
+            # assuming we have only one user named Administrator.
+            new_cache_refresh_time = r["Items"][0]['LastRefreshTime']
+        except (KeyError, IndexError):
+            print("The cache does not exist.")
+            return
+
+        print("New cache refresh time {} ".format(new_cache_refresh_time))
+        conn.close()
+
+        assert new_cache_refresh_time != old_cache_refresh_time, "The cache has not been refreshed."
+
+    except Exception as er:
+        print("Could not clear cache due to {}".format(er))
+        assert False, "Could not clear cache due to {}".format(er)
+
+
+@pytest.fixture
+def clear_pi_system_through_pi_web_api():
```

Comment:
If this fixture can be an alias to imported def to a python module, that would be great. That module should have a minimal argparser so that we can reuse the code and replace https://github.com/fledge-iot/fledge/blob/develop/tests/system/lab/verify_clean_pi.py

Reply:
Ok let's do that also in separate pull request.